resource/systemd/errors.go

```diff
@@ -20,6 +20,8 @@ import (
 	"github.com/hashicorp/go-multierror"
 )
 
+// MultiErrorAppend appends errors together into one skipping nil errors.
+// If all errors passed in are nil then it also returns nil
 func MultiErrorAppend(errs ...error) error {
```

Comment:
again, you shouldn't need this because you shouldn't be appending nil errors. Can you please fix that case and remove this code?